### PR TITLE
Rework `ParsedDataset` saving structure

### PR DESCRIPTION
**Pull request by lsrosa**
Testing scripts updated on `XAI`'s `ds_save_struct` branch

separate values copied from `DatasetWrap` from the inference's ones

add wrap for stacking `DatasetWrap` and parsed inference values saved on a `PersistentTensorDict` 

remove old `fine_tune()`

move transform from `DatasetWrap` to `parsedDataset`

organize tests

update release notes

### CODE REVIEW

File: peepholelib/datasets/CUB.py [hunk 1]
```diff
@@ -222,32 +221,32 @@ def __getitem__(self, idx):
...
-            _, W_new, H_new = img.shape
+        _, W_new, H_new = img.shape
 
-            scale_x = W_new / W_orig
-            scale_y = H_new / H_orig
+        scale_x = W_new / W_orig
+        scale_y = H_new / H_orig
 
-            bbox = torch.tensor([x * scale_x, y * scale_y, w  * scale_x, h * scale_y])
+        bbox = torch.tensor([x * scale_x, y * scale_y, w  * scale_x, h * scale_y])
```
Comment (lorenzoCapelli) [inline peepholelib/datasets/CUB.py]: shouldn't we comment the part related to bbox then

File: peepholelib/datasets/imagenetC.py [hunk 2]
```diff
@@ -4,12 +4,15 @@
...
 from torch.utils.data import Dataset
+from torchvision.transforms import ToTensor 
 
 from peepholelib.datasets.datasetWrap import DatasetWrap
 
 class CustomDS(Dataset):
-    def __init__(self, samples, synset_to_label, transform):
+    def __init__(self, samples, synset_to_label):
         Dataset.__init__(self)
+        self._to_tensor = ToTensor()
```
Comment (lorenzoCapelli) [inline peepholelib/datasets/imagenetC.py]: no need of `self._to_tensor = Compose([ToTensor(), Resize((224, 224))])` since those are images derived from ImageNet

Comment (lorenzoCapelli) [reply peepholelib/datasets/imagenetC.py]: sorry it was a question not a statement

File: peepholelib/datasets/CUB.py [hunk 3]
```diff
@@ -218,38 +217,38 @@ def __getitem__(self, idx):
...
+        parts_categorical = scaled_parts
+        ''' 
         sample = {
-            "image": img,
+            "image": self._to_tensor(img),
             "label": label,
-            "bbox": bbox,
+            #"bbox": bbox,
             **attributes_categorical,
             **parts_categorical
```
Comment (lorenzoCapelli) [inline peepholelib/datasets/CUB.py:246]: parts_categorical should be commented too, shouldnt it.

File: peepholelib/datasets/cifar10.py [hunk 4]
```diff
@@ -3,51 +3,54 @@
...
+            path (str): CIFAR10 download folder. If not already available, the dataset is downloaded in this folder. images.
+            augmentation (callable, optional): If provided, applied only to the training split.
+            train_ratio (float, optional): Fraction of training samples used for train (remainder goes to val).
+            seed (int, optional): Random seed used for deterministic train/val splitting.
         Returns:
             - a thumbs up
         '''
+        DatasetWrap.__init__(self, **kwargs)
 
         self.transform = kwargs.get('std_transform')
```
Comment (lorenzoCapelli) [inline peepholelib/datasets/cifar10.py]: Manca un `None`

File: peepholelib/models/model_wrap.py [hunk 5]
```diff
@@ -13,11 +13,11 @@
...
 class InputNormalizer(nn.Module):
 
-    def __init__(self, mean, std):
+    def __init__(self, mean, std, device):
         super(InputNormalizer, self).__init__()
 
-        self.register_buffer('mean', mean)
-        self.register_buffer('std', std)
+        self.mean = mean.to(device)
+        self.std = std.to(device)
```
Comment (lorenzoCapelli) [inline peepholelib/models/model_wrap.py:20]: Fai attenzione alla back prop

File: peepholelib/models/model_wrap.py [hunk 6]
```diff
@@ -151,6 +150,8 @@ def __call__(self, x):
         Returns:
             res (torch.tensor): the model output
         '''
+
+        x = self._normalizer(x)
```
Comment (lorenzoCapelli) [inline peepholelib/models/model_wrap.py:154]: check sul train con il normalizer

File: peepholelib/datasets/functional/transforms.py [hunk 7]
```diff
@@ -17,125 +36,94 @@
...
 
 vgg16_imagenet_augmentations = transforms.Compose([
     transforms.Resize((224, 224)),
     transforms.AutoAugment(policy=transforms.AutoAugmentPolicy.IMAGENET),
-    transforms.ToTensor(), 
     ])
 
 vgg16_svhn_augmentations = transforms.Compose([
     transforms.Resize((224, 224)), 
     transforms.AutoAugment(policy=transforms.AutoAugmentPolicy.SVHN),
```
Comment (lorenzoCapelli) [inline peepholelib/datasets/functional/transforms.py:57]:
```suggestion
vgg16_svhn_augmentations = transforms.Compose([
transforms.Resize((224, 224)), 
transforms.ToPILImage(),
transforms.AutoAugment(policy=transforms.AutoAugmentPolicy.SVHN), 
transforms.ToTensor()
```

Comment (lorenzoCapelli) [reply peepholelib/datasets/functional/transforms.py:57]: I would do like this only for augmentation

**Review by lorenzoCapelli — APPROVED** [review]
